scripts/core/program.yml

```diff
@@ -10,6 +10,67 @@ type: header
 desc: "Intel $OneApi Level-Zero Runtime APIs for Program"
 ordinal: "2"
 --- #--------------------------------------------------------------------------
+type: enum
+desc: "Program metadata property type."
+name: $x_program_metadata_type_t
+etors:
+    - name: UNKNOWN  # TODO: Is this necessary?
+      desc: "type is unknown."
+    - name: UINT32
+      desc: "type is a 32-bit integer."
+    - name: UINT64
+      desc: "type is a 32-bit integer."
+    - name: BYTE_ARRAY
+      desc: "type is a byte array."  # TODO: null-terminated?
```

Comment:
if the question is if we should ask users to provide the null-terminated character, I'd say yes.

Reply:
I should deleted this comment

Reply:
We actually have a `size` paremeter [here](https://github.com/oneapi-src/unified-runtime/pull/275/files#diff-49df2b979e9b956e061467dd2f8e1a6a07d5a2ac09f33acf90dac1603cb3ceefR54-R56) so null termination is not strictly necessary.